### PR TITLE
0.8

### PR DESCRIPTION
**Pull request by MeirionHughes**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,12 +42,12 @@
     "test": "gulp test"
   },
   "dependencies": {
-    "aurelia-binding": "^1.0.0-rc.1.0.2",
-    "aurelia-dependency-injection": "^1.0.0-rc.1.0.0",
-    "aurelia-pal-nodejs": "latest",
-    "aurelia-polyfills": "latest",
-    "aurelia-templating": "latest",
-    "aurelia-templating-binding": "latest",
+    "aurelia-binding": "1.0.0-rc.1.0.0",
```
Comment (atsu85) [inline package.json]: What was the reason to downgrade binding dependency?

File: source/ast.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    public children: ASTNode[];
+    public location: FileLoc;
+
+    constructor() {
+        this.parent = null;
+        this.children = [];
+        this.locals = [];
+        this.context = null;
+        this.location = null;
+    }
```
Comment (atsu85) [inline source/ast.ts]: I'd remove the constructor and initialize the defaults on the field declaration line instead

File: package.json (hunk 1 above)
Comment (MeirionHughes) [inline package.json]: locked everything when trying to find a bug; will unlock.

File: source/ast.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+export class ASTTextNode extends ASTNode {
+    public expression: InterpolationBindingExpression;
+
+    constructor() {
+        super();
+    }
+}
+
+export interface BehaviorInstruction {
```
Comment (atsu85) [inline source/ast.ts]: I guess it is worth documenting why BehaviorInstruction from aurelia-templating wasn't suitable

Comment (MeirionHughes) [inline source/ast.ts]: Waiting for aurelia team to update typings...

File: source/reflection.ts [hunk 4]
```diff
@@ -76,11 +76,62 @@ export class Reflection {
...
+        }
+    }
+
+    public resolveTypeElementType(node: ts.TypeElement): string {
+        if(!node)return null;
+        switch (node.kind) {
+            case ts.SyntaxKind.PropertySignature:
+                let prop = <ts.PropertySignature>node
+                return this.resolveTypeName(prop.type);
+            case ts.SyntaxKind.PropertySignature:
```
Comment (atsu85) [inline source/reflection.ts]: bug from copy-paste?

File: source/reflection.ts [hunk 5]
```diff
@@ -76,11 +76,62 @@ export class Reflection {
...
+                let meth = <ts.PropertySignature>node
+                return this.resolveTypeName(meth.type);
+            default:
+                console.log(ts.SyntaxKind[node.kind]);
```
Comment (atsu85) [inline source/reflection.ts]: i guess You either didn't plan to commit it, or the log message should be improved

File: source/rules/static-type-old.ts [hunk 6]
```diff
@@ -349,12 +349,11 @@ export class StaticTypeRule extends Rule {
             case ts.SyntaxKind.PropertySignature:
                 let prop = <ts.PropertySignature>node
                 return this.resolveTypeName(prop.type);
-            case ts.SyntaxKind.MethodSignature:
-                let meth = <ts.MethodSignature>node
+            case ts.SyntaxKind.PropertySignature:
```
Comment (atsu85) [inline source/rules/static-type-old.ts]: another copy-paste bug?

Comment (MeirionHughes) [inline source/rules/static-type-old.ts]: ah yeah, that entire file should be deleted. :D

**Pull request merged**